src/indra_cogex/analysis/source_targets_explanation.py

```diff
@@ -341,18 +347,57 @@ def shared_upstream_bioentities_from_targets(
         gene_ids=target_genes
     )
 
+    # Ensure upstream_df has proper CURIE format
+    if 'curie' in upstream_df.columns:
+        # Check if CURIEs have namespace prefix (like "hgnc:")
+        has_prefix = upstream_df['curie'].str.contains(':', na=False)
+
+        # For those without prefix, add appropriate namespace
+        if not all(has_prefix):
+            # Add HGNC namespace to numeric IDs that don't have a prefix
+            numeric_without_prefix = (~has_prefix) & upstream_df['curie'].str.match(r'^\d+$', na=False)
+            upstream_df.loc[numeric_without_prefix, 'curie'] = 'hgnc:' + upstream_df.loc[
+                numeric_without_prefix, 'curie']
+
+            # Log the corrections
+            prefix_count = numeric_without_prefix.sum()
+            if prefix_count > 0:
+                logger.info(f"Added namespace prefix to {prefix_count} CURIEs")
+
     # Find shared proteins
     shared_proteins = list(set(upstream_df["name"].values).intersection(
         set(stmts_by_protein_df["name"].values)))
 
     if shared_proteins:
+        # Get the entities data for these proteins
         shared_entities = upstream_df[upstream_df.name.isin(shared_proteins)]
-        logger.info("Found shared upstream bioentities")
+        logger.info(f"Found {len(shared_proteins)} shared upstream bioentities")
+
+        # Apply statistical filtering
+        if 'q' in shared_entities.columns:
+            significant_entities = shared_entities[shared_entities.q <= q_value_threshold]
+            logger.info(
+                f"After filtering, {len(significant_entities)} regulators are significant at q≤{q_value_threshold}")
+        else:
+            significant_entities = shared_entities
+
+        # Sort by significance
+        if 'q' in significant_entities.columns:
+            sorted_entities = significant_entities.sort_values('q')
```

Comment:
Instead of sorting here, we could add an argument to `_do_ora` that sets how to sort the values: https://github.com/gyorilab/indra_cogex/blob/main/src/indra_cogex/client/enrichment/discrete.py#L181